lms/djangoapps/grades/new/course_grade.py

```diff
@@ -52,6 +52,19 @@ def locations_to_weighted_scores(self):
                 locations_to_weighted_scores.update(subsection_grade.locations_to_weighted_scores)
         return locations_to_weighted_scores
 
+    @lazy
+    def grade_summary(self):
+        """
+        Helper function to extract the grade summary as calculated by the course's grader.
+        """
+        # Grading policy might be overriden by a CCX, need to reset it
+        self.course.set_grading_policy(self.course.grading_policy)
+        grade_summary = self.course.grader.grade(
+            self.subsection_grade_totals_by_format,
+            generate_random_scores=settings.GENERATE_PROFILE_SCORES
+        )
+        return grade_summary
```

Comment:
Nit no need for an intermediate `grade_summary` variable.  Can `return` directly from the above line.